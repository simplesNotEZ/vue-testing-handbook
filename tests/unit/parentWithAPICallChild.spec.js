import { shallowMount, mount } from "@vue/test-utils";
import ParentWithAPICallChild from "@/components/ParentWithAPICallChild.vue";
import ComponentWithAsyncCall from "@/components/ComponentWithAsyncCall.vue";

describe("ParentWithAPICallchild", () => {
  it.skip("renders with mount and does intialize API call", () => {
    const wrapper = mount(ParentWithAPICallChild);

    expect(wrapper.findComponent(ComponentWithAsyncCall).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  // replace the above test to stub the child component
  it("renders stubbed ComponentWithAsyncCall with mount and does initialize API call", () => {
    const wrapper = mount(ParentWithAPICallChild, {
      stubs: {
        ComponentWithAsyncCall: true
      }
    });

    expect(wrapper.findComponent(ComponentWithAsyncCall).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
