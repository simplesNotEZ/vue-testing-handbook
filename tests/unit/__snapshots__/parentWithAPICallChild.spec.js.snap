// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentWithAPICallchild renders stubbed ComponentWithAsyncCall with mount and does initialize API call 1`] = `<div class="stub"></div>`;
